providers/org/zenodo/normalizer.py

```diff
@@ -0,0 +1,569 @@
+
+
+class CreatorInstitution(Parser):
+    schema = 'Institution'
+
+    name = ctx.creatorName
+
+
+class CreatorOrganization(Parser):
+    schema = 'Organization'
+
+    name = ctx
+
+
+class Identifier(Parser):
+    base_url = Try(ctx['@schemeURI'])
+    url = Join(Concat(Try(ctx['@schemeURI']), Try(ctx['#text'])), joiner='/')
+
+
+class ThroughIdentifiers(Parser):
+    identifier = Delegate(Identifier, ctx)
+
+
+class ContributorPerson(Parser):
+    schema = 'Person'
+
+    suffix = ParseName(ctx.contributorName).suffix
+    family_name = ParseName(ctx.contributorName).last
+    given_name = ParseName(ctx.contributorName).first
+    additional_name = ParseName(ctx.contributorName).middle
+    identifiers = Map(Delegate(ThroughIdentifiers), Concat(Try(ctx, 'nameIdentifier')))
+
+    class Extra:
+        name_identifier = Try(
+            RunPython(
+                force_text,
+                ctx.nameIdentifier
+            )
+        )
+        name_identifier_scheme = Try(ctx.nameIdentifier['@nameIdentifierScheme'])
+        name_identifier_scheme_uri = Try(ctx.nameIdentifier['@schemeURI'])
+        contributor_type = Try(ctx.contributorType)
+
+
+class CreatorPerson(Parser):
+    schema = 'Person'
+
+    suffix = ParseName(ctx.creatorName).suffix
+    family_name = ParseName(ctx.creatorName).last
+    given_name = ParseName(ctx.creatorName).first
+    additional_name = ParseName(ctx.creatorName).middle
+    affiliations = Map(Delegate(Affiliation.using(entity=Delegate(CreatorOrganization))), Concat(Try(
+        RunPython(
+            force_text,
+            ctx.affiliation
+        )
+    )))
+    identifiers = Map(Delegate(ThroughIdentifiers), Concat(Try(ctx, 'nameIdentifier')))
+
+    class Extra:
+        name_identifier = Try(
+            RunPython(
+                force_text,
+                ctx.nameIdentifier
+            )
+        )
+        name_identifier_scheme = Try(ctx.nameIdentifier['@nameIdentifierScheme'])
+        name_identifier_scheme_uri = Try(ctx.nameIdentifier['@schemeURI'])
+
+
+class Contributor(Parser):
+    person = Delegate(ContributorPerson, ctx)
+    cited_name = Try(ctx.contributorName)
+    order_cited = ctx('index')
+
+
+class Creator(Parser):
+    schema = 'Contributor'
+
+    person = Delegate(CreatorPerson, ctx)
+    cited_name = Try(ctx.creatorName)
+    order_cited = ctx('index')
+
+
+class Funder(Parser):
+    community_identifier = Join(Concat(Try(ctx.nameIdentifier['@schemeURI']), Try(ctx.nameIdentifier['#text'])), joiner='/')
+
+    class Extra:
+        name = Try(ctx.contributorName)
+        name_identifier_scheme = Try(ctx.nameIdentifier['@nameIdentifierScheme'])
+        name_identifier_scheme_uri = Try(ctx.nameIdentifier['@schemeURI'])
+
+
+class Publisher(Parser):
+    name = ctx
+
+
+class Association(Parser):
+    pass
+
+
+class Tag(Parser):
+    name = ctx
+
+
+class ThroughTags(Parser):
+    tag = Delegate(Tag, ctx)
+
+
+class CreativeWork(Parser):
```

Comment:
Umm we could do that in the normalizer and route each one to a different class. Then have all of those classes like preprint, publication inherit from CreativeWork or something so that they will be labeled correctly. I think that's what I did elsewhere.